Algorithms/Graphs/MST/prim2.cpp

```diff
@@ -0,0 +1,99 @@
+
+/*  The following is an implementation of Prim's algorithm
+    using adjacency list representation of a graph. The data structure
+    min heap is used for this. There are, again, two implementations
+    in this, the min heap code can be written differently in a class, or
+    the priority_queue present in STL can be used. Here, it's the
+    latter.
+    Now, a priority_queue is essentially a max heap, but can be used
+    as given below to form a min heap. The loop statements execute
+    v+e times, in addition to log(v) time for adding to priority_queue.
+    Overall, O(v+e)*O(log(v)) = O((e+v)*log(v)) = O(e*log(v))
+    Since for a connected graph, v = O(e).
```

Comment:
Add an explanation for this

Reply:
I don't understand, for which part? For log(v)?

Reply:
Your claim v = O(E)

Reply:
e = O(v^2) for dense graphs, because e <= v*(v-1)/2 for any connected graph.
And e = O(v), or, v = O(e) for sparsely connected graphs.
I'm not sure if this is the correct explanation though.

Reply:
You can lookup on this in greater detail and then modify the code to add an explanation for the same.

Reply:
I did look it up, I think this is correct.